ledger-service/http-json/src/it/scala/http/WebsocketServiceIntegrationTest.scala

```diff
@@ -450,6 +450,80 @@ class WebsocketServiceIntegrationTest
       } yield resumes.foldLeft(1 shouldBe 1)((_, a) => a)
   }
 
+  "fetch multiple keys should work" in withHttpService { (uri, encoder, _) =>
+    def matches(expected: Seq[JsValue], actual: Seq[JsValue]): Boolean =
+      expected.length == actual.length && (expected, actual).zipped.forall {
+        case (exp, act) => matchesJs(exp, act)
+      }
+    def matchesJs(expected: spray.json.JsValue, actual: spray.json.JsValue): Boolean = {
+      import spray.json._
+      (expected, actual) match {
+        case (JsArray(expected), JsArray(actual)) =>
+          expected.length == actual.length && matches(expected, actual)
+        case (JsObject(expected), JsObject(actual)) =>
+          expected.keys.forall(k => matchesJs(expected(k), actual(k)))
+        case (JsString(expected), JsString(actual)) => expected == actual
+        case (JsNumber(expected), JsNumber(actual)) => expected == actual
+        case (JsBoolean(expected), JsBoolean(actual)) => expected == actual
+        case (JsNull, JsNull) => true
+        case _ => false
+      }
+    }
+    def waitFor[A](f: Future[A]): A = {
+      import scala.language.postfixOps
+      Await.result(f, 10 seconds)
+    }
+    def create(account: String): domain.ContractId = {
+      val r = waitFor(
+        postCreateCommand(accountCreateCommand(domain.Party("Alice"), account), encoder, uri))
+      assert(r._1.isSuccess)
+      getContractId(getResult(r._2))
+    }
+    def archive(id: domain.ContractId): Unit = {
+      val r =
+        waitFor(postArchiveCommand(domain.TemplateId(None, "Account", "Account"), id, encoder, uri))
+      val _ = assert(r._1.isSuccess)
+    }
+    val req =
+      """
+          |[{"templateId": "Account:Account", "key": ["Alice", "abc123"]},
+          | {"templateId": "Account:Account", "key": ["Alice", "def456"]}]
+          |""".stripMargin
+    val futureResults =
+      singleClientFetchStream(jwt, uri, req).via(parseResp).runWith(Sink.seq[JsValue])
+
+    val cid1 = create("abc123")
```

Comment:
Why are we using `Await.result` everywhere instead of chaining futures making use of the async test support in scalatest?

Reply:
My brain can't handle asyncness.

Reply:
It can definitely take some getting used to but for the most part, it turns out to be fairly nice in the end and I’d like to do this properly here. Here’s the diff to remove all the `Await.result`: https://gist.github.com/cocreature/d35367153a7331dc15cca4e5ea9098f0

Reply:
Strongly disagree there's anything proper (or better) about it, but you're the boss.

Reply:
The documentation for `Await.result` explains some of the reasons why it’s usually a bad idea https://www.scala-lang.org/api/current/scala/concurrent/Await$.html. I did run into exactly those issues in the past. Admittedly this isn’t as crucial in tests but avoiding it consistently is much less trouble than trying to make sure that it’s safe in a particular instance. So especially if the async version is hardly any longer or more complex like in this case, I’d like to choose that one.